packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts

```diff
@@ -33,114 +28,181 @@ type GetByViewerOptions = {
+        },
+        select: {
+          ...userEventTypeSelect,
+        },
+        orderBy: [
           {
-            upId: lightProfile.upId,
+            position: "desc",
           },
           {
-            where: {
-              teamId: null,
-            },
-            orderBy: [
-              {
-                position: "desc",
-              },
-              {
-                id: "asc",
-              },
-            ],
-          }
-        )
-      : [],
-  ]);
+            id: "asc",
+          },
+        ],
+      },
+    },
+  });
 
-  if (!profile) {
+  if (!unenrichedUser) {
     throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
   }
+  const user = UserRepository.enrichUserWithItsProfileBuiltFromUser({ user: unenrichedUser });
 
-  const memberships = profileMemberships.map((membership) => ({
+  const memberships = user.teams.map((membership) => ({
     ...membership,
     team: {
       ...membership.team,
       metadata: teamMetadataSchema.parse(membership.team.metadata),
     },
   }));
 
-  log.debug(
-    safeStringify({
-      profileMemberships,
-      profileEventTypes,
-    })
-  );
+  type UserEventTypes = (typeof user.eventTypes)[number];
+  type TeamEventTypeChildren = (typeof user.teams)[number]["team"]["eventTypes"][number];
 
-  type UserEventTypes = (typeof profileEventTypes)[number];
-  type TeamEventTypeChildren = NonNullable<(typeof profileEventTypes)[number]["team"]>["eventTypes"][number];
-  const mapEventType = async (eventType: UserEventTypes & Partial<TeamEventTypeChildren>) => ({
-    ...eventType,
-    safeDescription: eventType?.description ? markdownToSafeHTML(eventType.description) : undefined,
-    users: await Promise.all(
-      (!!eventType?.hosts?.length ? eventType?.hosts.map((host) => host.user) : eventType.users).map(
-        async (u) =>
-          await UserRepository.enrichUserWithItsProfile({
-            user: u,
-          })
-      )
-    ),
-    metadata: eventType.metadata ? EventTypeMetaDataSchema.parse(eventType.metadata) : null,
-    children: await Promise.all(
-      (eventType.children || []).map(async (c) => ({
-        ...c,
-        users: await Promise.all(
-          c.users.map(
-            async (u) =>
-              await UserRepository.enrichUserWithItsProfile({
-                user: u,
-              })
-          )
-        ),
-      }))
-    ),
-  });
+  const mapEventType = (eventType: UserEventTypes & Partial<TeamEventTypeChildren>) => {
+    const users = !!eventType?.hosts?.length ? eventType?.hosts.map((host) => host.user) : eventType.users;
+    return {
+      ...eventType,
+      safeDescription: eventType?.description ? markdownToSafeHTML(eventType.description) : undefined,
+      users: users.map((user) => UserRepository.enrichUserWithItsProfileBuiltFromUser({ user })),
+      metadata: eventType.metadata ? EventTypeMetaDataSchema.parse(eventType.metadata) : null,
```

Comment:
The change that fixed the type error seen in earlier PR https://github.com/calcom/cal.com/pull/13580/files#diff-5294731b960e76c72f504ce7e1381d51e5c054131976fe7f43bb38f485d76aa6R118